mkosi/__init__.py

```diff
@@ -3654,6 +3654,39 @@ def generate_btrfs(args: MkosiArgs, root: Path, label: str, for_cache: bool) ->
     return f
 
 
+def generate_xfs(args: MkosiArgs, root: Path, label: str, for_cache: bool) -> Optional[BinaryIO]:
+    if args.output_format != OutputFormat.gpt_xfs:
+        return None
+    if for_cache:
+        return None
+
+    @contextlib.contextmanager
+    def get_loopdev(f: BinaryIO) -> Iterator[Path]:
+        c = run(["losetup", "--find", "--show", f.name], stdout=PIPE, text=True)
+        l = Path(c.stdout.strip())
+        try:
+            yield l
+        finally:
+            run(["losetup", "--detach", l])
+
+    with complete_step("Creating xfs root file system…"):
+        f: BinaryIO = cast(
+            BinaryIO,
+            tempfile.NamedTemporaryFile(prefix=".mkosi-mkfs-xfs", dir=os.path.dirname(args.output))
+        )
+
+        f.truncate(args.root_size)
+        run(mkfs_xfs_cmd(label) + [f.name])
+
+        with tempfile.TemporaryDirectory(prefix=".mkosi-xfs-mnt", dir=os.path.dirname(args.output)) as td:
```

Comment:
We can just mount to `xfs` under `workspace(root)`, no need to use a separate temporary directory

Reply:
> We can just mount to `xfs` under `workspace(root)`, ...

I don't understand. Where am I supposed to mount the loopback device to copy the file system contents into?

Reply:
You can use `workspace(root) / "xfs"` as the directory to mount to instead of a temporary directory. Of course you'll have to create it first.

Reply:
Fixed